data/migrations/20211102125418_courses.js

```diff
@@ -1,60 +1,68 @@
 exports.up = (knex) => {
   return knex.schema
-    .createTable('course_types', function (table) {
-      table.increments('course_type_id');
-      table.text('description').notNullable();
-      table.string('subject').notNullable().unique();
-      table.specificType('prereqs', 'text ARRAY');
+    .createTable('programs', function (table) {
+      table.increments('program_id');
+      table.string('program_name').notNullable().unique();
+      table.text('program_description').notNullable();
     })
 
     .createTable('classes', (table) => {
       table.increments('class_id');
-      table.integer('size').notNullable();
-      table.integer('open_seats_remaining');
-
+      table.string('class_name').notNullable();
+      table.string('class_description').notNullable();
+      table.specificType('days_of_week', 'text ARRAY');
+      table.integer('max_size').notNullable();
+      table.integer('min_age').notNullable();
+      table.integer('max_age').notNullable();
       table
         .integer('instructor_id')
+        .notNullable()
         .unsigned()
         .notNullable()
         .references('instructor_id')
         .inTable('instructors')
         .onDelete('CASCADE');
 
       table
-        .integer('course_type_id')
+        .integer('program_id')
+        .notNullable()
         .unsigned()
         .notNullable()
-        .references('course_type_id')
-        .inTable('course_types')
+        .references('program_id')
+        .inTable('programs')
         .onDelete('CASCADE');
       table.time('start_time').notNullable();
       table.time('end_time').notNullable();
       table.date('start_date').notNullable();
       table.date('end_date').notNullable();
       table.string('location').notNullable();
+      table.integer('number_of_sessions').notNullable();
     })
-    .createTable('instructors_course_types', (table) => {
-      table.increments('instructors_course_types_id');
+    .createTable('instructors_program_types', (table) => {
```

Comment:
Makes perfect sense for the joining table! Now we can track which instructors are available to teach which programs.